.golangci.yaml

```diff
@@ -0,0 +1,55 @@
+linters-settings:
+  errcheck:
+    check-type-assertions: true
+    check-blank: true
+  govet:
+    check-shadowing: true
+  gofmt:
+    simplify: false
+  golint:
+    min-confidence: 0
+  gocyclo:
+    min-complexity: 10
+  maligned:
+    suggest-new: true
+  dupl:
+    threshold: 100
+  goconst:
+    min-len: 3
+    min-occurrences: 3
+  misspell:
+    locale: US
+  goimports:
+    local-prefixes: github.com/rahul23/trivy
+
+linters:
+  disable-all: true
+  enable:
+    - structcheck
+    - ineffassign
+    - typecheck
+    - govet
+    - errcheck
```

Comment:
I think we can drop `errcheck`.

Reply:
I can drop it, but then we are enabling developers to ignore errors, I think we should keep it, at least the developer should log the error, it will help us if we miss errors while development.

Reply:
It is better to have it, but we have to add `// nolint: errcheck` to all `defer` and some tests. It's a bit bothering me. Do we have another way besides `nolint`?

Reply:
I already skipped the test and mock files in the linter config so this is not required in tests, Another way would be to have a common package for system functions and instead of calling file.Open and closing in defer we will call the custom package function to get file reader.

Reply:
@knqyf263 I have added the exception for obvious cases and now we don't need to add nolint in general cases.